pkg/order/mempool/mempool_impl.go

```diff
@@ -248,5 +276,75 @@ func (mpi *mempoolImpl) processCommitTransactions(state *ChainState) {
 	}
 	mpi.logger.Debugf("Replica %d removes batches in mempool, and now there are %d non-batched txs,"+
 		"priority len: %d, parkingLot len: %d, batchedTx len: %d, txHashMap len: %d", mpi.localID, mpi.txStore.priorityNonBatchSize,
-		 mpi.txStore.priorityIndex.size(), mpi.txStore.parkingLotIndex.size(), len(mpi.txStore.batchedTxs), len(mpi.txStore.txHashMap))
+		mpi.txStore.priorityIndex.size(), mpi.txStore.parkingLotIndex.size(), len(mpi.txStore.batchedTxs), len(mpi.txStore.txHashMap))
+}
+
+func (mpi *mempoolImpl) GetTimeoutTransactions(rebroadcastDuration time.Duration) [][]*pb.Transaction {
+	// all the tx whose live time is less than lowBoundTime should be rebroadcast
+	mpi.logger.Debugf("------- Start gathering timeout txs, ttl index len is %d -----------", mpi.txStore.ttlIndex.index.Len())
+	currentTime := time.Now().UnixNano()
+	if currentTime < mpi.txStore.earliestTimestamp+rebroadcastDuration.Nanoseconds() {
+		// if the latest incoming tx has not exceeded the timeout limit, then none will be timeout
+		return [][]*pb.Transaction{}
+	}
+
+	txList := make([]*pb.Transaction, 0)
```

Comment:
It is seems that txlist is not needed, and shardTxList can be divided by timeoutItems.